src/features/game/expansion/placeable/lib/collisionDetection.ts

```diff
@@ -345,163 +343,85 @@ export function isWithinAOE(
-      }
-    }
-    // AoE for the Laurie the Chuckle Crow
-    if (AOEItemName === "Laurie the Chuckle Crow") {
-      const topLeft = {
-        x: AOEItemCoordinates.x - 1,
-        y: AOEItemCoordinates.y - AOEItem.height,
-      };
-
-      const bottomRight = {
-        x: AOEItemCoordinates.x + 1,
-        y: AOEItemCoordinates.y - AOEItemDimensions.height - 2,
-      };
+  const isWithinDistance = (
+    dx: number,
+    dy: number,
+    distance: number
+  ): boolean => {
+    return Math.abs(dx) <= distance && Math.abs(dy) <= distance;
+  };
 
-      if (
-        effectItem.x >= topLeft.x &&
-        effectItem.x <= bottomRight.x &&
-        effectItem.y <= topLeft.y &&
-        effectItem.y >= bottomRight.y
-      ) {
-        return true;
-      }
+  switch (AOEItemName) {
+    case "Basic Scarecrow":
+    case "Scary Mike":
+    case "Laurie the Chuckle Crow": {
+      return isWithinRectangle(
+        { x: x - 1, y: y - height, height, width },
+        { x: x + 1, y: y - height - 2, height, width }
+      );
     }
 
-    if (AOEItemName === "Queen Cornelia") {
-      const topLeft = {
-        x: AOEItemCoordinates.x - 1,
-        y: AOEItemCoordinates.y + 1,
-      };
-
-      const bottomRight = {
-        x: AOEItemCoordinates.x + AOEItemDimensions.width,
-        y: AOEItemCoordinates.y - AOEItemDimensions.height,
-      };
-
-      if (
-        effectItem.x >= topLeft.x &&
-        effectItem.x <= bottomRight.x &&
-        effectItem.y <= topLeft.y &&
-        effectItem.y >= bottomRight.y
-      ) {
-        return true;
-      }
+    case "Emerald Turtle":
+    case "Tin Turtle": {
+      const dxTurtle = x - effectItem.x;
+      const dyTurtle = y - effectItem.y;
+      return (
+        isWithinDistance(dxTurtle, dyTurtle, 1) &&
+        (dxTurtle !== 0 || dyTurtle !== 0)
+      );
     }
-
-    // AoE surrounding the turtle
-    if (AOEItemName === "Emerald Turtle" || AOEItemName === "Tin Turtle") {
-      const dx = Math.abs(AOEItemCoordinates.x - effectItem.x);
-      const dy = Math.abs(AOEItemCoordinates.y - effectItem.y);
-
-      if (dx <= 1 && dy <= 1 && (dx != 0 || dy != 0)) {
-        return true;
-      }
+    case "Sir Goldensnout":
+    case "Bale": {
+      const dxRect = effectItem.x - x;
+      const dyRect = effectItem.y - y;
+      return (
+        dxRect >= -1 && dxRect <= width && dyRect <= 1 && dyRect >= -height
+      );
     }
 
-    // AoE surrounding the Sir Goldensnout
-    if (AOEItemName === "Sir Goldensnout") {
-      const dx = effectItem.x - AOEItemCoordinates.x;
-      const dy = effectItem.y - AOEItemCoordinates.y;
-
-      if (
-        dx >= -1 &&
-        dx <= AOEItemDimensions.width && // Covers the width of the Goldensnout and one tile around it
-        dy <= 1 &&
-        dy >= -AOEItemDimensions.height // Covers the height of the Goldensnout and one tile around it
-      ) {
-        return true;
-      }
+    case "Queen Cornelia": {
+      return isWithinRectangle(
```

Comment:
(Consider) If `isWithinDistance` had an optional `dy` parameter, then Queen Cornelia, Sir GoldenSnout, and Bale could be expressed using `isWithinDistance`.